### PR TITLE
Catch server error in issue comments

### PR DESCRIPTION
**Pull request by ericboucher**
Temporary alleviation of #128

### CODE REVIEW

File: tap_github/repository_streams.py [hunk 1]
```diff
@@ -891,7 +891,16 @@ def get_records(self, context: Optional[Dict] = None) -> Iterable[Dict[str, Any]
             self.logger.debug(f"No comments detected. Skipping '{self.name}' sync.")
             return []
 
-        return super().get_records(context)
+        # Getting issue comments is sometimes too expensive on large repos and results server errors.
+        # To avoid crashing the tap completely, we wrap the get_records in a try/except.
+        try:
+            return super().get_records(context)
+        except ValueError:
+            self.logger.warning(
```
Comment (laurentS) [inline tap_github/repository_streams.py:899]: Is it worth filtering on the error message itself (if at all possible), so as to ignore only these specific errors, and not ones we actually want to crash on?

File: tap_github/repository_streams.py [hunk 2]
```diff
@@ -891,7 +891,16 @@ def get_records(self, context: Optional[Dict] = None) -> Iterable[Dict[str, Any]
             self.logger.debug(f"No comments detected. Skipping '{self.name}' sync.")
             return []
 
-        return super().get_records(context)
+        # Getting issue comments is sometimes too expensive on large repos and results server errors.
```
Comment (laurentS) [inline tap_github/repository_streams.py:894]:
```suggestion
        # Getting issue comments is sometimes too expensive on large repos and results in server errors.
```

Comment (laurentS) [reply tap_github/repository_streams.py:894]: Maybe worth linking to the relevant github api docs?

**Review by laurentS — APPROVED** [review]

File: tap_github/repository_streams.py (hunk 2 above)
Comment (ericboucher) [reply tap_github/repository_streams.py:894]: I could not really find a github doc for that, did you?

File: tap_github/repository_streams.py (hunk 1 above)
Comment (ericboucher) [reply tap_github/repository_streams.py]: I added a string check, not sure if it's the best way though

**Comment by laurentS** [comment]
@ericboucher do we still need this PR?

**Comment by tayloramurphy** [comment]
@ericboucher what's the status on this PR?

**Comment by ericboucher** [comment]
I think we can close for now

**Pull request closed without merging**